Issue60/color picker state in provider

ramirezStefano opened the pull request:
Steve here! followed up the ticket requirements and specifications, ready for review!

ramirezStefano committed:
feat: add color state to provider and wirder up with onColorChange

re #60

ramirezStefano committed:
refactor: set and spreading color prop to the color state/

re #60

eduardo-vargas commented on src/providers/SignatureProvider.tsx:
```diff
@@ -26,6 +28,7 @@ const createSignatureProviderState = () => {
     const [linkedin, setLinkedin] = useState<InputElementOptions>(getDefaultInputElementOptions("linkedin"));
     const [phone, setPhone] = useState<InputElementOptions>(getDefaultInputElementOptions("phone"));
     const [isDarkMode, setIsDarkMode] = useLocalStorageState(localStorageKeys.isDarkMode, false);
+    const [color, setColor] = useState<string>();
```
We shouldn't need to have a dedicated useState for the color since we're adding it to the complex state object

eduardo-vargas commented on src/components/UsernameInput/UsernameInput.tsx:
```diff
@@ -13,19 +14,30 @@ interface UsernameInputProps extends Omit<TextStyleProps, "onColorChange"> {
 
 const UsernameInput: FC<UsernameInputProps> = ({ state, onInputChange, label }) => {
-    const handleButtonChange = (options: { isBold?: boolean; isItalics?: boolean }) => {
+    const { setColor } = useSignatureContext();
```
We want to keep this component completely generic, with this we would be tightly coupling it to the provider 

We should leverage the same `onInputChange` method that updates the complex state behind the scenes, and it should behave very similar to the `handleInputChange` method 

Something like 

```javascript
onInputChange({ variant: state.variant, color: color })
```

then inside the provider, we do something like 

```javascript
               newOptions.style = {
                    fontWeight: newOptions.isBold ? "bold" : "normal",
                    fontStyle: newOptions.isItalics ? "italic" : "normal",
                    color: newOptions.color ? newOptions.color.toString("hex") : "black"
                };
```

Notice how we do the `toString` at the very end of the process to maintain the integrity of the `Color` interface, if we did it at the `onChange` level, by the time it gets to the provider it would be a string making it hard to manipulate as a color.

eduardo-vargas commented on src/components/UsernameInput/UsernameInput.tsx:
```diff
@@ -13,19 +14,30 @@ interface UsernameInputProps extends Omit<TextStyleProps, "onColorChange"> {
-    const handleColorChange = (color: Color) => {
-        // eslint-disable-next-line no-console
-        console.log(color.toFormat("hex"));
+    const updateProviderColorState = (value: Color) => {
+        const strColor = value.toString("hex");
+        setColor(strColor);
+        return strColor;
+    };
+
+    const handleColorChange = (colorValue: Color) => {
+        const strColorValue = updateProviderColorState(colorValue);
+        console.log(strColorValue); //working setting value on state. missing wire up to
+
+        console.log(colorValue.toFormat("hex"));
```
See comment above, we can clean this part a bit 
```suggestion
    
    
    
    
    

    const handleColorChange = (colorValue: Color) => {
        const strColorValue = onInputChange({ variant: state.variant, color: colorValue });
        

        
```

eduardo-vargas commented on src/providers/SignatureProvider.tsx:
```diff
@@ -7,6 +7,7 @@ interface InputElementOptions {
     isBold: boolean;
     isItalics: boolean;
     style: CSSProperties;
+    color?: string;
```
Lets make this of type `Color`

```javascript 
import { Color } from "@react-types/color";
```

eduardo-vargas commented on src/providers/SignatureProvider.tsx:
```diff
@@ -15,6 +16,7 @@ const getDefaultInputElementOptions = (type: InputElementOptions["variant"]): In
     isBold: false,
     isItalics: false,
     style: { fontWeight: "normal", fontStyle: "normal" },
+    color: "",
```
Lets make the default color `black`

```suggestion
    color: parseColor("#000000"),
```

```javascript
import { parseColor } from "@react-stately/color";
```

ramirezStefano committed:
refactor: remove color state and leveraged the onInputChange

Use of the spread operator to set the remaining css color value to each variant field.

re #60

ramirezStefano commented:
Yellow and good morning! Finished the implementation using the provider onUpdate function to spread the color css value to the style object, code reviewed and tested, good times! 🚀

eduardo-vargas reviewed (approved):
Love it!! 🚀 
<img width="335" alt="image" src="https://github.com/evargast/signature-generator/assets/52969178/4ee9420a-f026-43d6-9098-a26affb244b4"><img width="256" alt="image" src="https://github.com/evargast/signature-generator/assets/52969178/e4e78b03-d307-49a0-aaab-0f7cdbecd74d">

ramirezStefano commented:
All conflicts resolved and going with the merge YOLO again! Finger crossed for no blue smoke! 🔥
